app/BUILD.bazel

```diff
@@ -245,6 +253,50 @@ ALTERED_VIEWS = [
  )
 for file in ALTERED_VIEWS]
 
+# Source files and genrule targets for databinding_adapters library.
+DATABINDING_ADAPTER_FILES = [
+    "src/main/java/org/oppia/app/databinding/EditTextBindingAdapters.java",
+    "src/main/java/org/oppia/app/databinding/GuidelineBindingAdapters.java",
+    "src/main/java/org/oppia/app/databinding/MarginBindingAdapters.java",
+    "src/main/java/org/oppia/app/databinding/StateAssemblerMarginBindingAdapters.java",
+    "src/main/java/org/oppia/app/databinding/StateAssemblerPaddingBindingAdapters.java",
+    "src/main/java/org/oppia/app/databinding/ViewBindingAdapters.java",
+    "update_src/main/java/org/oppia/app/databinding/DrawableBindingAdapters",
+    "update_src/main/java/org/oppia/app/databinding/ImageViewBindingAdapters",
+    "update_src/main/java/org/oppia/app/databinding/TextViewBindingAdapters",
+]
+
+# TODO(#1617): Remove genrules post-gradle
+# Adapter files altered by genrules to be excluded in the app library.
+ALTERED_DATABINDING_ADAPTERS = [
+    "src/main/java/org/oppia/app/databinding/DrawableBindingAdapters.java",
+    "src/main/java/org/oppia/app/databinding/ImageViewBindingAdapters.java",
+    "src/main/java/org/oppia/app/databinding/TextViewBindingAdapters.java",
+]
+
+'''
+Genrule that alters both resource import statements and class names for adapter files in order to
+account for altered package and file name.
+'''
+[genrule(
+     name = "update_" + file[0:-5],
+     srcs = [file],
+     outs = [file[0:-5] + "_updated.java"],
+     cmd = '''
+     cat $(SRCS) |
+     sed 's/import org.oppia.app.R/import org.oppia.app.impl.adapters.R/g' |
+     sed 's/BindingAdapters {/BindingAdapters_updated {/g' > $(OUTS)
+     ''',
+ )
+for file in ALTERED_DATABINDING_ADAPTERS]
+
+# Source files for recyclerview_adapters library.
+RECYCLERVIEW_ADAPTER_FILES = [
+    "src/main/java/org/oppia/app/recyclerview/RecyclerViewBindingAdapter.java"
+]
+
+BINDING_ADAPTER_FILES = DATABINDING_ADAPTER_FILES + RECYCLERVIEW_ADAPTER_FILES
```

Comment:
These different names are a bit confusing (``binding`` vs ``databinding``)--please rename or consolidate to clarify.

Reply:
I altered quite a few names - let me know if you think it helps!

Reply:
This is much clearer--thanks!